rollout/trafficrouting/istio/istio.go

```diff
@@ -57,30 +58,50 @@ const (
 	invalidCasting = "Invalid casting: field '%s' is not of type '%s'"
 )
 
-func (patches virtualServicePatches) patchVirtualService(httpRoutes []interface{}) error {
+func (patches virtualServicePatches) patchVirtualService(httpRoutes []interface{}, tlsRoutes []interface{}) error {
 	for _, patch := range patches {
-		route, ok := httpRoutes[patch.routeIndex].(map[string]interface{})
-		if !ok {
-			return fmt.Errorf(invalidCasting, "http[]", "map[string]interface")
-		}
-		destinations, ok := route["route"].([]interface{})
-		if !ok {
-			return fmt.Errorf(invalidCasting, "http[].route", "[]interface")
-		}
-		destination, ok := destinations[patch.destinationIndex].(map[string]interface{})
-		if !ok {
-			return fmt.Errorf(invalidCasting, "http[].route[].destination", "map[string]interface")
-		}
-		destination["weight"] = float64(patch.weight)
+		if patch.routeType == "http" {
+			route, ok := httpRoutes[patch.routeIndex].(map[string]interface{})
+			if !ok {
+				return fmt.Errorf(invalidCasting, "http[]", "map[string]interface")
+			}
+			destinations, ok := route["route"].([]interface{})
+			if !ok {
+				return fmt.Errorf(invalidCasting, "http[].route", "[]interface")
+			}
+			destination, ok := destinations[patch.destinationIndex].(map[string]interface{})
+			if !ok {
+				return fmt.Errorf(invalidCasting, "http[].route[].destination", "map[string]interface")
+			}
+			destination["weight"] = float64(patch.weight)
+
+			destinations[patch.destinationIndex] = destination
+			route["route"] = destinations
+			httpRoutes[patch.routeIndex] = route
+		} else if patch.routeType == "tls" {
+			route, ok := tlsRoutes[patch.routeIndex].(map[string]interface{})
+			if !ok {
+				return fmt.Errorf(invalidCasting, "tls[]", "map[string]interface")
+			}
+			destinations, ok := route["route"].([]interface{})
+			if !ok {
+				return fmt.Errorf(invalidCasting, "tls[].route", "[]interface")
+			}
+			destination, ok := destinations[patch.destinationIndex].(map[string]interface{})
+			if !ok {
+				return fmt.Errorf(invalidCasting, "tls[].route[].destination", "map[string]interface")
+			}
+			destination["weight"] = float64(patch.weight)
```

Comment:
Line 94-97 are duplicate to line 76-79. Please consider refactoring it to out of if...else

Reply:
Done!